test/end2end_test.go

```diff
@@ -193,6 +193,16 @@ func (s *testServer) HalfDuplexCall(stream testpb.TestService_HalfDuplexCallServ
 
 const tlsDir = "testdata/"
 
+func TestDialTimeout(t *testing.T) {
+	conn, err := grpc.Dial("localhost:0", grpc.WithTimeout(5*time.Microsecond))
```

Comment:
On Mon, Mar 2, 2015 at 1:53 AM, Wes <notifications@github.com> wrote:

> In test/end2end_test.go
> <https://github.com/grpc/grpc-go/pull/88#discussion_r25587275>:
>
> > @@ -193,6 +193,16 @@ func (s *testServer) HalfDuplexCall(stream testpb.TestService_HalfDuplexCallServ
> >
> >  const tlsDir = "testdata/"
> >
> > +func TestDialTimeout(t *testing.T) {
> > +	conn, err := grpc.Dial("localhost:0", grpc.WithTimeout(5*time.Microsecond))
>
> Hrm, I'm not sure I need to start up a server here, since the easiest way
> to test this timeout is to attempt connecting to an unopened socket.
>
> So, how to go about getting a socket we're sure is unopened--in the flaky
> case, perhaps some other test leaves one open on accident? I've implemented
> a simplistic open/close/test-with-recently-closed-address logic, but I'm
> not really happy with it. If you have a better idea, I'm all ears :) but
> it's 2AM here and I'll have to take another look in the morning.
>
Thanks for the effort. My idea is to connect to a non-existing dns name:
conn, err := grpc.Dial("Non-Existing.Server:80",
grpc.WithTimeout(time.Millisecond))
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/88/files#r25587275>.
>